src/ostree.cc

```diff
@@ -88,15 +88,15 @@ bool Ostree::addRemote(OstreeRepo *repo, const std::string &remote, const std::s
 
 #include "ostree-1/ostree.h"
 
-OstreePackage::OstreePackage(const std::string &ecu_serial_in, const std::string &ref_name_in,
-                             const std::string &branch_name_in, const std::string &refhash_in,
-                             const std::string &desc_in, const std::string &treehub_in)
-    : ecu_serial(ecu_serial_in),
-      ref_name(ref_name_in),
-      branch_name(branch_name_in),
-      refhash(refhash_in),
-      description(desc_in),
-      pull_uri(treehub_in) {}
+OstreePackage::OstreePackage(const std::string &ref_name_in, const std::string &refhash_in, const std::string &desc_in,
+                             const std::string &treehub_in)
+    : ref_name(ref_name_in), refhash(refhash_in), description(desc_in), pull_uri(treehub_in) {
+  std::size_t pos = ref_name.find_last_of("-");
+  if (pos == std::string::npos) throw std::runtime_error("Invalid refname");
```

Comment:
I'd like to get rid of this requirement that the name in targets.json has a specific form--but maybe this can wait for a future PR